x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/predicate/Negatable.java

```diff
@@ -6,9 +6,9 @@
  */
 package org.elasticsearch.xpack.ql.expression.predicate;
 
-import org.elasticsearch.xpack.ql.expression.function.scalar.ScalarFunction;
+import org.elasticsearch.xpack.ql.expression.Expression;
 
-public interface Negatable<T extends ScalarFunction> {
+public interface Negatable<T extends Expression> {
```

Comment:
Made Negatable accept any kind of expression not just functions. This way Not can be negated without special handling.